src/components/PatientForm.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { FunctionComponent, useState, useContext, useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Patient } from '../types'
+import { TokenContext } from '../context/token'
+import { ErrorContext, ErrorType } from '../context/error'
+
+import { strict as assert } from 'assert'
+
+import Button from 'react-bootstrap/Button'
+import Row from 'react-bootstrap/Row'
+import { updatePatient } from '../services/api'
+
+interface PatientFormProps {
+  patient: Patient
+}
+
+interface Field {
+  value: any
+  error?: string
+  isValid: boolean
+}
+
+interface Form {
+  lastname: Field
+  firstname: Field
+  address: Field
+  city: Field
+  zipcode: Field
+
+}
+
+const PatientForm: FunctionComponent<PatientFormProps> = ({ patient }) => {
+  const { token } = useContext(TokenContext)
+  const { addError } = useContext(ErrorContext)
+
+  const [form, setForm] = useState<Form>({
+    lastname: { value: patient.lastname, isValid: true },
+    firstname: { value: patient.firstname, isValid: true },
+    address: { value: patient.address, isValid: true },
+    city: { value: patient.city, isValid: true },
+    zipcode: { value: patient.zip_code, isValid: true }
+  })
+
+  const history = useNavigate()
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const fieldName: string = e.target.name
+    const fieldValue: string = e.target.value
+    const newField = { [fieldName]: { value: fieldValue } }
+    setForm({ ...form, ...newField })
+  }
+
+  const addErrorCallback = useCallback(
+    (error: ErrorType) => addError(error),
+    // eslint-disable-next-line
+    []
+  )
+
+  const onUpdate = async (): Promise<void> => {
+    assert(token)
+    try {
+      await updatePatient(token, patient)
+    } catch (error) {
+      addErrorCallback({ body: 'Error updating patient' })
+    }
+  }
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>): Promise<void> => {
+    e.preventDefault()
+    patient.firstname = form.firstname.value
+    patient.lastname = form.lastname.value
+    patient.address = form.address.value
+    patient.city = form.city.value
+    patient.zip_code = form.zipcode.value
+    await onUpdate()
+    history(`/patients/${patient.id}`)
+  }
+
+  return (
+    <form onSubmit={async e => await handleSubmit(e)}>
+      <Row>
+        <div className='col s-12 m-8 offset-m-2'>
+          <div className='card hoverable'>
```

Comment:
Maybe we don't actually need a card here as it kind of overload the final rendering.
I mean we have extra borders that overlaps with the form control ones.
Maybe we should follow the same layout as the profile page to keep it consistent